src/containers/load-button.jsx

```diff
@@ -49,26 +48,34 @@ class LoadButton extends React.Component {
     setFileInput (input) {
         this.fileInput = input;
     }
+    renderFileInput () {
+        return (
+            <input
+                accept=".sb2,.sb3"
+                ref={this.setFileInput}
+                style={{display: 'none'}}
+                type="file"
+                onChange={this.handleChange}
+            />
+        );
+    }
     render () {
         if (this.state.loadingError) throw new Error(`Failed to load project: ${this.state.errorMessage}`);
         const {
-            closeLoadingState, // eslint-disable-line no-unused-vars
-            openLoadingState, // eslint-disable-line no-unused-vars
-            vm, // eslint-disable-line no-unused-vars
+            /* eslint-disable no-unused-vars */
+            children,
+            closeLoadingState,
+            openLoadingState,
+            vm,
+            /* eslint-enable no-unused-vars */
             ...props
         } = this.props;
-        return (
-            <LoadButtonComponent
-                inputRef={this.setFileInput}
-                onChange={this.handleChange}
-                onClick={this.handleClick}
-                {...props}
-            />
-        );
+        return this.props.children(this.renderFileInput, this.handleClick, props);
```

Comment:
I'm not sure I fully understand what's happening here... is it just a way of passing arguments that the children of this component can use?

Reply:
Comments would have helped, sorry :/  This is the piece that makes the HOC-replacement above work. This component expects its "children" to be a function. Then it calls that function with arguments that would have been props had it been a HOC.  So while it is a component, you can use it to enhance any component you like directly within the render function. With a HOC you have to wrap a component class ahead of time to use it.